File: strongtify_client/src/types/genre.ts
import { Album } from "./album";
import { Song } from "./song";

export type Genre = {
    id: string;
    name: string;
    alias: string;
    imageUrl?: string;
}

export type GenreDetail = {
    id: string;
    createdAt: string;
    updatedAt: string;
    name: string;
    alias: string;
    description?: string;
    imageUrl?: string;
    songs?: Song[];
    albums?: Album[];
}

export type CudGenreResponse = {
    id: string;
    name: string;
    alias: string;
    imageUrl?: string;
    description?: string;
}

export type CreateUpdateGenreRequest = {
    name: string;
    description?: string;
    image?: File;
}

export type GenreDetailQuery = {
    songLimit?: number;
    albumLimit?: number;
}

File: strongtify_mobile_app/lib/models/genre/genre.dart
class Genre {
  final String id;
  final String name;
  final String alias;
  final String? imageUrl;

  Genre({
    required this.id,
    required this.name,
    required this.alias,
    this.imageUrl,
  });

  factory Genre.fromMap(Map<String, dynamic> data) {
    return Genre(
      id: data['id'],
      name: data['name'],
      alias: data['alias'],
      imageUrl: data['imageUrl'],
    );
  }
}
